src/app/middlewares/checkToken.ts

```diff
@@ -0,0 +1,21 @@
+import { NextFunction, Request, Response } from "express";
+import { verify } from "jsonwebtoken";
+import { JWT_SECRET } from "../../config";
+import { getTokenFromRequestHeader } from "../../utils";
+import { Authorization, Forbidden } from "../errors";
+
+export function checkToken(req: Request, _: Response, next: NextFunction) {
+  const token = getTokenFromRequestHeader(req);
+
+  if (!token) {
+    throw new Authorization("token is required");
+  }
+
+  try {
+    verify(token, JWT_SECRET);
+  } catch (error) {
+    throw new Forbidden("invalid token");
```

Comment:
At this place, you should process all kinds of errors, which can happen, send them to the client with the right HTTP status code. 

Possible errors: https://github.com/auth0/node-jsonwebtoken#errors--codes

Reply:
fixed